C/plugins/north/OMF/linkdata.cpp

```diff
@@ -0,0 +1,299 @@
+/*
+ * Fledge OSIsoft OMF interface to PI Server.
+ *
+ * Copyright (c) 2022 Dianomic Systems
+ *
+ * Released under the Apache 2.0 Licence
+ *
+ * Author: Mark Riddoch
+ */
+
+#include <utility>
+#include <iostream>
+#include <string>
+#include <cstring>
+#include <omf.h>
+#include <OMFHint.h>
+#include <logger.h>
+#include "string_utils.h"
+#include <datapoint.h>
+
+#include <iterator>
+
+#include <omflinkeddata.h>
+
+using namespace std;
+/**
+ * OMFLinkedData constructor, generates the OMF message containing the data
+ *
+ * @param reading           Reading for which the OMF message must be generated
+ * @param AFHierarchyPrefix Unused at the current stage
+ * @param hints             OMF hints for the specific reading for changing the behaviour of the operation
+ *
+ */
+string OMFLinkedData::processReading(const Reading& reading, const string&  AFHierarchyPrefix, OMFHints *hints)
+{
+	string outData;
+	bool changed;
+
+
+	string assetName = reading.getAssetName();
+	// Apply any TagName hints to modify the containerid
+	if (hints)
+	{
+		const std::vector<OMFHint *> omfHints = hints->getHints();
+		for (auto it = omfHints.cbegin(); it != omfHints.cend(); it++)
+		{
+			if (typeid(**it) == typeid(OMFTagNameHint))
+			{
+				assetName = (*it)->getHint();
+				Logger::getLogger()->info("Using OMF TagName hint: %s", assetName.c_str());
```

Comment:
There are 3 _info_ messages in this method. Would it be more appropriate to make them _debug_ messages?